lib/plugins.js

```diff
@@ -0,0 +1,109 @@
+"use strict";
+
+let { loadExtension, abort, repr } = require("./util");
+
+// common plugins included for convenience
+let DEFAULTS = [{
+	key: "js",
+	bucket: "scripts",
+	plugin: "faucet-pipeline-js"
+}, {
+	key: "sass",
+	bucket: "styles",
+	plugin: "faucet-pipeline-sass"
+}, {
+	key: "static",
+	bucket: "static",
+	plugin: "faucet-pipeline-static"
+}];
+let DEFAULT_KEYS = DEFAULTS.reduce((memo, plugin) => {
+	memo.add(plugin.key);
+	return memo;
+}, new Set());
+let BUCKETS = new Set(["static", "scripts", "styles", "markup"]);
```

Comment:
This list is duplicated, somewhat implicitly, in `lib/index.js`:
https://github.com/faucet-pipeline/faucet-pipeline-core/blob/22620e5b7e0f03d676784d53fe5a8f3375e96e1b/lib/index.js#L29-L33
It might be nice to make that connection more obvious by having that code iterate over `BUCKETS` instead, except
a) order matters, so `Set` _might_ no longer be appropriate ([MDN suggests it's fine](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Description) though),
b) programmatically creating a promise chain there seems like a questionable idea.

WDYT; worth it?

Reply:
As the churn for this list will probably be zero (those buckets are basically just the categories of things that a Web server delivers, not sure why this would change), I vote for keeping it like this, as duplication doesn't hurt when things don't change.

Reply:
Agreed. 👌